tuning/data/data_processors.py

```diff
@@ -393,23 +393,10 @@ def process_dataset_configs(
         self, dataset_configs: List[DataSetConfig], **kwargs
     ) -> Union[Dataset, IterableDataset]:
         train_dataset = None
+        state = PartialState()
```

Comment:
Is partial state always available? do we need to check if it gets initialised properly?

Reply:
Could you also add here the reasoning you mentioned in the PR on why Partial state is preferred and possibly link with the HF docs to mention that its the recommended pattern. 
you can point to the code too https://github.com/huggingface/trl/blob/main/trl/trainer/sft_trainer.py#L367